pdblp/pdblp.py

```diff
@@ -424,3 +424,175 @@ def stop(self):
+        while (self.session.tryNextEvent()):
+            pass
+
+        request = self._beqs_create_req(screen_name, screen_type, group, language_id)
+
+        cid = None
+        if asof_date is not None:
+            overrides = request.getElement("overrides")
+            ovrd = overrides.appendElement()
+            ovrd.setElement("fieldId", "PiTDate")
+            ovrd.setElement("value", asof_date.strftime('%Y%m%d'))
+            cid = blpapi.CorrelationId(asof_date)
+
+        logging.debug("Sending Request:\n %s" % request)
+        # Send the request
+        self.session.sendRequest(request, correlationId=cid)
+        data = []
+        # Process received events
+        while (True):
+            # We provide timeout to give the chance for Ctrl+C handling:
+            ev = self.session.nextEvent(500)
+            data = self._beqs_parse_event(data, ev)
+
+            if ev.eventType() == blpapi.Event.RESPONSE:
+                # Response completely received, so we could exit
+                break
+
+        return data
+
+    def _beqs_parse_event(self, data, ev):
+        for msg in ev:
+            logging.debug("Message Received:\n %s" % msg)
+            corrID = msg.correlationIds()[0].value()
+            fldData = msg.getElement('data').getElement('securityData')
+            for i in range(fldData.numValues()):
+                ticker = (fldData.getValue(i).getElement("security").getValue())  # NOQA
+                reqFldsData = (fldData.getValue(i).getElement('fieldData'))
+                for e in reqFldsData.elements():
+                    fld = str(e.name())
+                    # this is for dealing with requests which return arrays
+                    # of values for a single field
+                    if reqFldsData.getElement(fld).isArray():
+                        lrng = reqFldsData.getElement(fld).numValues()
+                        for k in range(lrng):
+                            elms = (reqFldsData.getElement(fld).getValue(k).elements())  # NOQA
+                            # if the elements of the array have multiple
+                            # subelements this will just append them all
+                            # into a list
+                            for elm in elms:
+                                data.append([ticker, fld, elm.getValue(), corrID])
+                    else:
+                        val = reqFldsData.getElement(fld).getValue()
+                        data.append([ticker, fld, val, corrID])
+        return data
+
+    def _beqs_create_req(self, screen_name, screen_type, group, language_id):
+        request = self.refDataService.createRequest('BeqsRequest')
+        request.set('screenName', screen_name)
+        request.set('screenType', screen_type)
+        request.set('Group', group)
+        request.set('languageId', language_id)
+        return request
+
+    def beqs_hist(self, screen_name, screen_type='PRIVATE', group='General', language_id='ENGLISH',
```

Comment:
`freq` parameter here is unused and should be removed. `longdata` parameter should be documented.